src/java.base/share/classes/java/io/Console.java

```diff
@@ -214,6 +227,24 @@ public String readln(String prompt) {
         throw newUnsupportedOperationException();
     }
 
+    /**
+     * Reads a single line of text from this console.
+     *
+     * @throws IOError
+     *         If an I/O error occurs.
+     *
+     * @return  A string containing the line read from the console, not
+     *          including any line-termination characters, or {@code null}
+     *          if an end of stream has been reached without having read
+     *          any characters.
+     *
```

Comment:
Should we specify (both here and in the `IO.readln()`) what is returned if only the line-terminator character is read from the console? A brief look at the implementation seems to suggest that we return an empty `String`.

Reply:
I'm not an expert in this field, but It is pretty common to exclude line terminator from readln string in different languages. Therefore, It should be intuitive that if nothing was present in line, the result of method will be string with noting, i.e. empty string. The method could benefit from some sort of semantic annotation like NonNull, but I guess modern IDEs do pretty good job on inferring such things anyway.

PS: Also I feel like for person that never wrote code in their life "line terminator" would sound like some sort of error, maybe the better wording should be used in IO specifically

Reply:
I also find the text to be fairly clear.